### PR TITLE
fix(progress): turn twMerge true to fix custom styling

### LINKED ISSUE #2153: [BUG] - why the twMerge is false in progress.ts
### NextUI Version

2.2.9

### Describe the bug

the twMerge is false in progress.ts. then cannot customise Progress slots without twMerge

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

if we customise base in Progress
      <Progress
        ref={ref}
        classNames={{
          ...{
            base: 'flex flex-row-reverse',
          },
          ...classNames,
        }}
        label={`${current} of ${total}`}
        value={current}
        maxValue={total}
        {...props}
      />

### Expected behavior

expecting the original flex-col will be replaced by flex-row-reverse. but in reality it is like 'flex-col flex-row-reverse'.
i searched the source code. it is caused by     twMerge: false in progress.ts, which is odd because this is the only twMerge: false in the whole source code

### Screenshots or Videos

_No response_

### Operating System Version

macos 14.0 (23A344)

### Browser

Chrome

### PR DESCRIPTION
**Pull request by caebwallace**
Closes #2153

## 📝 Description

> Allow slot customisation by turning parameter `twMerge` of the `progress` theme to true.

image

## ⛳️ Current behavior (updates)

> Progress component can not be customised as default className override custom classNames

image

## 🚀 New behavior

> Use twMerge to apply customisation

image

## Summary by CodeRabbit

- **New Features**
	- Introduced `CustomSlots` for more flexible customization in progress components.
- **Bug Fixes**
	- Fixed an issue with custom slot styling not properly merging with Tailwind CSS in progress components.

### CODE REVIEW

File: .changeset/short-trainers-dance.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@nextui-org/progress": major
+"@nextui-org/theme": major
```
Comment (wingkwong) [inline .changeset/short-trainers-dance.md:3]: use `patch` instead of `major`

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: .changeset/short-trainers-dance.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@nextui-org/progress": patch
```
Comment (wingkwong) [inline .changeset/short-trainers-dance.md:2]: we don't need to add a changeset for stories changes. L2 can be removed.

Comment (caebwallace) [reply .changeset/short-trainers-dance.md]: you're right! Done!

**Comment by caebwallace** [comment]
Hi! Just checking in : everything okay with the code in that PR ?
Tell me if not :)

File: .changeset/short-trainers-dance.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@nextui-org/theme": patch
+---
+
+Fix custom slot styling with twMerge true
```
Comment (wingkwong) [inline .changeset/short-trainers-dance.md:5]: can you include the issue number here

`Fix custom slot styling with twMerge true (# 2153)`

Comment (caebwallace) [reply .changeset/short-trainers-dance.md]: Good! It's done

**Review by wingkwong — APPROVED** [review]

**Pull request merged**